[Linux] build error

cloned the repo and configured it. when running make I get the following error:

```
[ 23%] Building CXX object libopenage/CMakeFiles/libopenage.dir/event/demo/main.cpp.o
/home/user/openage/libopenage/event/demo/gui.cpp: In member function ‘void openage::event::demo::Gui::init()’:
/home/user/openage/libopenage/event/demo/gui.cpp:136:25: error: format not a string literal and no format arguments [-Werror=format-security]
  136 |                 mvprintw(row++, col, c);
      |                 ~~~~~~~~^~~~~~~~~~~~~~~
/home/user/openage/libopenage/event/demo/gui.cpp: In member function ‘void openage::event::demo::Gui::draw(const std::shared_ptr<openage::event::demo::PongState>&, const time_t&)’:
/home/user/openage/libopenage/event/demo/gui.cpp:209:25: error: format not a string literal and no format arguments [-Werror=format-security]
  209 |                 mvprintw((6 + msg_i), state->display_boundary[0]/2 + 10, msg.c_str());
      |                 ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 23%] Building CXX object libopenage/CMakeFiles/libopenage.dir/event/demo/physics.cpp.o

```
Then the build fails. OS is debian testing and gcc 11

Hello, I have the same error on Arch Linux, GCC 11.1.0 and Clang 13.0.1

```
[ 25%] Building CXX object libopenage/CMakeFiles/libopenage.dir/gamedata/tech_dummy.cpp.o
/home/fabio/Dev/Github/PKGBUILD/openage-git/src/openage/libopenage/event/demo/gui.cpp:136:24: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
                mvprintw(row++, col, c);
                                     ^
/home/fabio/Dev/Github/PKGBUILD/openage-git/src/openage/libopenage/event/demo/gui.cpp:136:24: note: treat the string as an argument to avoid this
                mvprintw(row++, col, c);
                                     ^
                                     "%s", 
/home/fabio/Dev/Github/PKGBUILD/openage-git/src/openage/libopenage/event/demo/gui.cpp:209:60: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
                mvprintw((6 + msg_i), state->display_boundary[0]/2 + 10, msg.c_str());
                                                                         ^~~~~~~~~~~
/home/fabio/Dev/Github/PKGBUILD/openage-git/src/openage/libopenage/event/demo/gui.cpp:209:60: note: treat the string as an argument to avoid this
                mvprintw((6 + msg_i), state->display_boundary[0]/2 + 10, msg.c_str());
                                                                         ^
                                                                         "%s", 
[ 25%] Building CXX object libopenage/CMakeFiles/libopenage.dir/gamedata/terrain_dummy.cpp.o

```

--- a/buildsystem/codecompliance/__main__.py
+++ b/buildsystem/codecompliance/__main__.py
@@ -1,4 +1,4 @@
-# Copyright 2014-2021 the openage authors. See copying.md for legal info.
+# Copyright 2014-2022 the openage authors. See copying.md for legal info.
 
 """
 Entry point for the code compliance checker.
@@ -10,6 +10,8 @@ import os
 import shutil
 import subprocess
 import sys
+
+from .util import log_setup
 
 
 def parse_args():
@@ -51,6 +53,11 @@ def parse_args():
     cli.add_argument("--fix", action="store_true",
                      help=("try to automatically fix the found issues"))
 
+    cli.add_argument("-v", "--verbose", action="count", default=0,
+                     help="increase program verbosity")
+    cli.add_argument("-q", "--quiet", action="count", default=0,
+                     help="decrease program verbosity")
+
     args = cli.parse_args()
     process_args(args, cli.error)
 
@@ -65,6 +72,9 @@ def process_args(args, error):
     """
     # this method is very flat; artificially nesting it would be bullshit.
     # pylint: disable=too-many-branches
+
+    # set up log level
+    log_setup(args.verbose - args.quiet)
 
     if args.fast or args.all:
         # enable "fast" tests

--- a/buildsystem/codecompliance/authors.py
+++ b/buildsystem/codecompliance/authors.py
@@ -1,10 +1,14 @@
-# Copyright 2014-2021 the openage authors. See copying.md for legal info.
+# Copyright 2014-2022 the openage authors. See copying.md for legal info.
 
 """
 Checks whether all authors are properly listed in copying.md.
 """
 
 import re
+
+import logging
+
+from .util import Strlazy
 
 
 def deobfuscate_email(string):
@@ -35,7 +39,7 @@ def get_author_emails_copying_md():
     """
     with open("copying.md", encoding='utf8') as fobj:
         for line in fobj:
-            match = re.match("^.*\\|[^|]*\\|[^|]*\\|([^|]*)\\|.*$", line)
+            match = re.match("^.*\\|[^|]*\\|[^|]*\\|([^|]+)\\|.*$", line)
             if not match:
                 continue
 
@@ -43,8 +47,9 @@ def get_author_emails_copying_md():
             if 'à' in email:
                 email = deobfuscate_email(email)
 
-            if '@' not in email:
-                continue
+            if not any(email.startswith(prefix) for prefix in ("E-Mail", "-" * 15))\
+               and '@' not in email:
+                raise Exception(f"no @ or à was found in email: {email}")
 
             yield email
 
@@ -83,7 +88,12 @@ def find_issues():
                      '.qml')
 
     copying_md_emails = set(get_author_emails_copying_md())
+    logging.debug("scanned authors in copying.md:\n%s",
+                  Strlazy(lambda: f"{chr(10).join(sorted(copying_md_emails))}"))
+
     git_shortlog_emails = set(get_author_emails_git_shortlog(relevant_exts))
+    logging.debug("scanned authors from git shortlog:\n%s",
+                  Strlazy(lambda: f"{chr(10).join(sorted(git_shortlog_emails))}"))
 
     # look for git emails that are unlisted in copying.md
     for email in git_shortlog_emails - copying_md_emails:

--- a/buildsystem/codecompliance/util.py
+++ b/buildsystem/codecompliance/util.py
@@ -1,9 +1,10 @@
-# Copyright 2014-2021 the openage authors. See copying.md for legal info.
+# Copyright 2014-2022 the openage authors. See copying.md for legal info.
 
 """
 Some utilities.
 """
 
+import logging
 import os
 
 
@@ -11,6 +12,44 @@ SHEBANG = "#!/.*\n(#?\n)?"
 
 FILECACHE = {}
 BADUTF8FILES = set()
+
+
+def log_setup(setting, default=1):
+    """
+    Perform setup for the logger.
+    Run before any logging.log thingy is called.
+
+    if setting is 0: the default is used, which is WARNING.
+    else: setting + default is used.
+    """
+
+    levels = (logging.ERROR, logging.WARNING, logging.INFO,
+              logging.DEBUG, logging.NOTSET)
+
+    factor = clamp(default + setting, 0, len(levels) - 1)
+    level = levels[factor]
+
+    logging.basicConfig(level=level, format="[%(asctime)s] %(message)s")
+    logging.captureWarnings(True)
+
+
+def clamp(number, smallest, largest):
+    """ return number but limit it to the inclusive given value range """
+    return max(smallest, min(number, largest))
+
+
+class Strlazy:
+    # pylint: disable=too-few-public-methods
+    """
+    to be used like this: logging.debug("rolf %s", strlazy(lambda: do_something()))
+    so do_something is only called when the debug message is actually printed
+    do_something could also be an f-string.
+    """
+    def __init__(self, fun):
+        self.fun = fun
+
+    def __str__(self):
+        return self.fun()
 
 
 def has_ext(fname, exts):

--- a/copying.md
+++ b/copying.md
@@ -139,7 +139,8 @@ _the openage authors_ are:
 | Jens Nyman                  | nymanjens                   | nymanjens dawt nj à gmail dawt com                |
 | Deepak Dinesh               | deepak                      | d.deepakdinesh13 à gmail dawt com                 |
 | Damien Lejay                | dlejay                      | lejay à paracompact dawt space                    |
-| Talha Aamir                | sarcxd                      | sarcxd à gmail dawt com                    |
+| Talha Aamir                 | sarcxd                      | sarcxd à gmail dawt com                           |
+| Matthias Geiger             | CountOmega                  | matthias dawt geiger1024 à outlook dawt com       |
 
 If you're a first-time committer, add yourself to the above list. This is not
 just for legal reasons, but also to keep an overview of all those nicknames.

--- a/libopenage/event/demo/gui.cpp
+++ b/libopenage/event/demo/gui.cpp
@@ -1,4 +1,4 @@
-// Copyright 2016-2019 the openage authors. See copying.md for legal info.
+// Copyright 2016-2022 the openage authors. See copying.md for legal info.
 
 #include "gui.h"
 
@@ -133,7 +133,7 @@ void Gui::init() {
 	int row = (y - buffer.size()) / 2;;
 	int col = (x - colwidth) / 2;
 	for (const auto &c : buffer) {
-		mvprintw(row++, col, c);
+		mvprintw(row++, col, "%c", c);
 	}
 
 	attroff(COLOR_PAIR(COLOR_DEBUG));
@@ -206,7 +206,7 @@ void Gui::draw(const std::shared_ptr<PongState> &state, const curve::time_t &now
 	// show log
 	int msg_i = 0;
 	for (auto & msg : this->log_msgs) {
-		mvprintw((6 + msg_i), state->display_boundary[0]/2 + 10, msg.c_str());
+		mvprintw((6 + msg_i), state->display_boundary[0]/2 + 10, "%s", msg.c_str());
 		msg_i += 1;
 	}
 

